vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/WalletService.kt

```diff
@@ -111,45 +139,101 @@ class WalletService(
         val clock: Clock = Clock.System,
     )
 
+    /**
+     * Pass in the URL provided by the Credential Issuer,
+     * which may contain a direct [CredentialOffer] or a URI pointing to it.
+     */
+    suspend fun parseCredentialOffer(input: String): KmmResult<CredentialOffer> = catching {
+        kotlin.runCatching {
```

Comment:
`catching { runCatching`?

Reply:
You're right, but imho changing this all the way through would pollute this PR.

Reply:
all the way through is not the goal for this PR, but nesting it like that just seems off

Reply:
catching directly inside catchig still seems fishy

Reply:
my bad. looking at the full functions helps. perfectly fine this way